web/src/components/MapView.tsx

```diff
@@ -28,19 +34,92 @@ const layerStyle: LayerProps = {
   paint: {}
 }
 
+const renderPopup = (coordinates: Position, description: string): ReactElement => {
+  return (
+    <StyledPopup
+      longitude={coordinates[0]}
+      latitude={coordinates[1]}
+      closeButton={false}
+      closeOnClick={false}
+      anchor='bottom'>
+      <div dangerouslySetInnerHTML={{ __html: description }} />
+    </StyledPopup>
+  )
+}
+
+const getPopUpDescription = (properties: GeoJsonProperties): string => {
+  return `<div><strong>${properties?.title}</strong></div>`
+}
+
+const getCursor = ({ isHovering, isDragging }: { isHovering: boolean; isDragging: boolean }): Property.Cursor => {
+  return isDragging ? 'grabbing' : isHovering ? 'pointer' : 'default'
+}
+
 interface MapViewProps {
-  featureCollection: FeatureCollection
+  featureCollection: FeatureCollection<Point, GeoJsonProperties>
 }
 
 const MapView: React.FunctionComponent<MapViewProps> = (props: MapViewProps): ReactElement => {
-  const [viewport, setViewport] = useState(defaultViewportConfig)
+  const [viewport, setViewport] = useState<MapViewViewport>(defaultViewportConfig)
+  const [showPopup, togglePopup] = React.useState<boolean>(false)
+  const [currentPoi, setCurrentPoi] = React.useState<Feature<Point, GeoJsonProperties> | null>(null)
+
   const { featureCollection } = props
+  const query = new URLSearchParams(useLocation().search)
+  const queryId = Number(query.get(mapParam))
+
+  const getCurrentPoi = useCallback(
+    (
+      features: FeatureCollection<Point, GeoJsonProperties>,
+      queryId: number
+    ): Feature<Point, GeoJsonProperties> | undefined => {
+      return featureCollection.features.find(feature => feature.properties?.id === queryId)
+    },
+    [featureCollection.features]
+  )
+
+  useEffect(() => {
+    const currentPoi: Feature<Point, GeoJsonProperties> | undefined = getCurrentPoi(featureCollection, queryId)
```

Comment:
Explicit typing not needed here
```suggestion
    const currentPoi = getCurrentPoi(featureCollection, queryId)
```